runtime/gc_vlhgc/WriteOnceCompactor.cpp

```diff
@@ -1243,7 +1243,11 @@ void
 MM_WriteOnceCompactor::doStackSlot(MM_EnvironmentVLHGC *env, J9Object *fromObject, J9Object **slotPtr, J9StackWalkState *walkState, const void *stackLocation)
 {
 	if (isHeapObject(*slotPtr)) {
+		Assert_MM_validStackSlot(MM_StackSlotValidator(MM_StackSlotValidator::COULD_BE_FORWARDED, *slotPtr, stackLocation, walkState).validate(env));
```

Comment:
we probably don't need it here. validation will be done earlier through marking
I don't see it done for regular threads being done either